File: Excel/455 Challenge.R
library(tidyverse)
library(readxl)
library(numbers)

input = read_excel("Excel/455 Anti perfect numbers.xlsx", range = "A1:A10")
test  = read_excel("Excel/455 Anti perfect numbers.xlsx", range = "B1:B5")

is_antiperfect = function(number) {
  divisors = divisors(number) 
  divisors = divisors[-length(divisors)]
  reversed_divisors = map(divisors, ~str_c(rev(str_split(.x, "")[[1]]), collapse = "")) %>%
    as.numeric()
  sum_rev_div = sum(reversed_divisors)
  return(sum_rev_div == number)
}

result = input %>%
  mutate(is_antiperfect = map_lgl(Numbers, is_antiperfect)) %>%
  filter(is_antiperfect) %>%
  select(`Expected Answer` = Numbers)

identical(result, test)
# [1] TRUE

File: Excel/400-499/455/455 Challenge.py
import pandas as pd
import numpy as np

input = pd.read_excel("455 Anti perfect numbers.xlsx", sheet_name="Sheet1", usecols="A")
test = pd.read_excel("455 Anti perfect numbers.xlsx", sheet_name="Sheet1", usecols="B", nrows = 4)

def is_antiperfect(number):
    divisors = [i for i in range(1, number) if number % i == 0]
    reversed_divisors = [int(str(i)[::-1]) for i in divisors]
    sum_rev_div = sum(reversed_divisors)
    return sum_rev_div == number

input["is_antiperfect"] = input["Numbers"].apply(is_antiperfect)
result = input[input["is_antiperfect"]]["Numbers"].rename("Expected Answer").astype("int64").reset_index(drop=True)

print(result.equals(test["Expected Answer"])) # True
